qiskit/circuit/quantumcircuit.py

```diff
@@ -66,6 +66,7 @@
     import qiskit  # pylint: disable=cyclic-import
     from qiskit.transpiler.layout import TranspileLayout  # pylint: disable=cyclic-import
     from qiskit.quantum_info.operators.base_operator import BaseOperator
+    from qiskit.quantum_info.states.statevector import Statevector  # pylint: disable=cyclic-import
```

Comment:
Minor, but the import should be from the highest level it's exported from (i.e. `from qiskit.quantum_info import Statevector`), because `QuantumCircuit` should only be using the public API of `quantum_info`, and `.states.statevector` is internal detail.  We only ought to drill further into the package structure when it matters for resolution of import cycles (but ideally we shouldn't have import cycles across public interface boundaries of packages at all).

Reply:
Ah yes, that's a good reminder! This import is now not needed anymore though, as the `prepare_state` method is now still monkey-patched.